### PR TITLE
[script] add option `OTBR_RADIO_URL` to set the radio URL

### PR DESCRIPTION
**Pull request by zhanglongxia**
When the RCP bus type is set to SPI, the otbr-agent's arguments also need to be updated to specify the SPI interface. This commit adds an option `OTBR_RADIO_URL` to configure the otbr-agent's radio URL.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ $ git submodule update --init --recursive
 At the root of the repository:
 
 ```
-$ REFERENCE_PLATFORM=(nrf52840|efr32mg12|ncs|none) REFERENCE_RELEASE_TYPE=(1.2|1.3)  [SD_CARD=/dev/...] [OTBR_RCP_BUS=(UART|SPI)] [IN_CHINA=(0|1)] ./script/make-reference-release.bash
+$ REFERENCE_PLATFORM=(nrf52840|efr32mg12|ncs|none) REFERENCE_RELEASE_TYPE=(1.2|1.3)  [SD_CARD=/dev/...] [OTBR_RCP_BUS=(UART|SPI)] [IN_CHINA=(0|1)] [OTBR_AGENT_OPTS='-I wpan0 -B wlan0 "spinel+hdlc+uart:///dev/ttyUSB0" trel://wlan0'] ./script/make-reference-release.bash
```
Comment (superwhd) [inline README.md:20]:
```suggestion
$ REFERENCE_PLATFORM=(nrf52840|efr32mg12|ncs|none) REFERENCE_RELEASE_TYPE=(1.2|1.3)  [SD_CARD=/dev/...] [OTBR_RCP_BUS=(UART|SPI)] [IN_CHINA=(0|1)] [OTBR_AGENT_OPTS='-I wpan0 -B wlan0 spinel+hdlc+uart:///dev/ttyUSB0 trel://wlan0'] ./script/make-reference-release.bash
```

Seems unnecessary to add quotes

File: README.md [hunk 2]
```diff
@@ -33,6 +33,10 @@ China so that you can save time while installing software dependencies.
 `OTBR_RCP_BUS` sets the RCP hardware interface type of Spinel on otbr-agent.
 If this variable is not specified, the interface type is set to `UART` by default.
 
+`OTBR_AGENT_OPTS` sets the otbr-agent's arguments. If this variable is not specified,
+the otbr-agent's argument is set to
+`-I wpan0 -B eth0 "spinel+hdlc+uart:///dev/ttyACM0" trel://eth0` by default.
```
Comment (superwhd) [inline README.md:38]: ditto

**Review by superwhd — APPROVED** [review]
LGTM :+1:

File: script/otbr-setup.bash [hunk 3]
```diff
@@ -120,6 +121,16 @@ deb-src http://mirrors.tuna.tsinghua.edu.cn/raspbian/raspbian/ buster main non-f
...
     fi
 }
+
+set_otbr_agent_opts()
+{
+    local otbr_default_settings_file="/home/pi/repo/ot-br-posix/src/agent/otbr-agent.default.in"
+    if [ -n "$OTBR_AGENT_OPTS" ]; then
+        sed -i '/OTBR_AGENT_OPTS=.*/d' "${otbr_default_settings_file}"
+        echo "OTBR_AGENT_OPTS=${OTBR_AGENT_OPTS}" >> "${otbr_default_settings_file}"
+    fi
```
Comment (librasungirl) [inline script/otbr-setup.bash:131]: direct replacement seems a bit tricky

File: README.md [hunk 4]
```diff
@@ -17,7 +17,7 @@ $ git submodule update --init --recursive
 At the root of the repository:
 
 ```
-$ REFERENCE_PLATFORM=(nrf52840|efr32mg12|ncs|none) REFERENCE_RELEASE_TYPE=(1.2|1.3)  [SD_CARD=/dev/...] [OTBR_RCP_BUS=(UART|SPI)] [IN_CHINA=(0|1)] ./script/make-reference-release.bash
+$ REFERENCE_PLATFORM=(nrf52840|efr32mg12|ncs|none) REFERENCE_RELEASE_TYPE=(1.2|1.3)  [SD_CARD=/dev/...] [OTBR_RCP_BUS=(UART|SPI)] [IN_CHINA=(0|1)] [OTBR_AGENT_OPTS='-I wpan0 -B wlan0 spinel+hdlc+uart:///dev/ttyUSB0 trel://wlan0'] ./script/make-reference-release.bash
```
Comment (librasungirl) [inline README.md:20]: How about config `INFRA_IF_NAME` to change the infrastructure interface?
may add option to specify RADIO_URL or RADIO_DEV_URL for flexible thread device configuration?

File: script/otbr-setup.bash (hunk 3 above)
Comment (zhanglongxia) [reply script/otbr-setup.bash:131]: The premise of setting the `OTBR_AGENT_OPTS` is that the user understands the function of the parameters. From this point, I think it is reasonable to directly use the new configuration to replace the default configuration.

File: README.md (hunk 4 above)
Comment (zhanglongxia) [reply README.md:20]: We can add `INFRA_IF_NAME`  and `RADIO_URL `, but it makes the configuration more restricted and complex.  Thinking about adding new arguments in the future, or we want to change other arguments, we still can use the `OTBR_AGENT_OPTS` to sets the new argument without adding new configuration variables to the code. So I think setting the `OTBR_AGENT_OPTS` directly is more flexible than setting multiple configurations.

**Review by librasungirl — COMMENTED** [review]
The PR seems beyond the scope of the original problem to solve "When the RCP bus type is set to SPI, the otbr-agent's arguments also need to be updated to specify the SPI interface. This commit adds an option OTBR_AGENT_OPTS to configure the otbr-agent's arguments"

If only for specifying spi interface, I may prefer a bit directly add an option to specify the serial device (/dev/ttyACM0 or /dev/spidev0.0) and hide the details of different url prefixes `spinel+hdlc+uart` or `spinel+spi` from users.

Please update either the subject/PR, or the problem for match
Please add a caution for `OTBR_AGENT_OPTS` option and may document some typical usage samples (e.g. per radio url prefix) for easier adoption by the community if you prefer current approach

**Review by librasungirl — APPROVED** [review]
LGTM 👍

**Pull request merged**